io/src/main/java9/io/smallrye/common/io/classloader/ClassDefiner.java

```diff
@@ -0,0 +1,14 @@
+package io.smallrye.common.io.classloader;
+
+import java.lang.invoke.MethodHandles;
+
+public class ClassDefiner {
+    public static Class<?> defineClass(Class<?> parent, String className, byte[] classBytes) {
+        try {
+            MethodHandles.Lookup lookup = MethodHandles.privateLookupIn(parent, MethodHandles.lookup());
```

Comment:
This won't work unless this exact class has access to the target parent class.  You'd have to pass in a trusted Lookup from the caller AFAICT otherwise you risk getting access exceptions in various situations.

Reply:
Hey @dmlloyd was about to message you about this PR :)

I didn't invest too much in this, or fully tested it. I just wanted to check with you if you agree with the general strategy on how to support the `defineClass`.

Reply:
Meaning if you agree with the MR Jar and have different impls for a `defineClass`.

Reply:
If the strategy is fine, I'll clean up, fix and add more tests.

Reply:
I agree with the general approach.  Maybe it doesn't belong in "io" though because it's not I/O?  It might be worth its own module specific to class loading.

Reply:
Yes, I thought about this as well, but I wasn't sure if it was worth to add its own module. I'll move it. Thanks!